### DISCUSSION TITLE
sjtuthesis 如何在英文扉页的"公开论文"的方框处画勾，以及日期如何居中

### DISCUSSION THREAD

**Discussion by CrazyElements** [sjtug/SJTUThesis#1034] [提问]
![image](https://github.com/sjtug/SJTUThesis/assets/105211051/1d1d52d1-50ec-4151-8d42-ba42d9d05145)
我自己尝试了一下 checkdlist，但效果不好；也不知道该如何将日期居中显示。

**Comment by NemoYuan2008** [comment]
你说的日期居中是指学位论文原创性声明中作者签名和日期居中吗？如果是这样的话，在导言区添加以下代码
```latex
\EditInstance{sjtu}{copyright/orig/sign}{
    format = \zihao{4}\fixedlineskip{31.2bp}\centering
}
```

打勾的话，直接用 `$\checkmark$` 吧

**Reply by CrazyElements** [reply]
谢谢你，我确实是需要在原创性声明处居中显示日期（标题错写为英文扉页了）

能请您详细说说怎么画勾吗? 我只在某个文件下找到了*_white_square_tl 表示方框，我在其后添加\checkmark，发现只是在方框后画勾

**Reply by NemoYuan2008** [reply]
你可以将对应选项中的 `\c__sjtu_symbol_white_square_tl` 直接替换为 `\,\checkmark`（无需我上面写的 `$`），即可将方框改成打勾。其中 `\,` 是为了打出一点间距，以便勾和下面的方框对齐

**Reply by CrazyElements** [reply]
在 sjtu-name-thesis-ah.def 修改为 ｛ \,\checkmark ｝后，只出现勾，但没有，没有方框啊

**Reply by CrazyElements** [reply]
还有原创性声明处，我是想让数字在空白处居中，比如 2024 在冒号和年间空白处的正中间，而不是说签名和日期需要在整行的中间（不好意思，之前表达得有问题）

**Reply by LogCreative** [reply]
将

https://github.com/sjtug/SJTUThesis/blob/0eff2175eb573293a1d274262a2325c6ab5c41c0/texmf/tex/latex/sjtutex/sjtu-name-thesis-zh.def#L98-L100

改为

```latex
    日期： 2024 \hfil 年
           \hfil 5 \hfil 月
           \hfil 11 \hfil 日
```

**Comment by AlexaraWu** [comment]
方框处画勾可以用 `\llap`，把两个符号叠起来。
```latex
\c__sjtu_symbol_white_square_tl \llap { \ensuremath { \checkmark } }
```

**Comment by CrazyElements** [comment]
@NemoYuan2008, @LogCreative, @AlexaraWu, 谢谢各位，我的问题解决了
